src/typescript-worker/util.ts

```diff
@@ -130,3 +130,15 @@ export const fileExtension = (path: string): string => {
     ? ''
     : path.slice(lastDotIdx + 1);
 };
+
+/**
+ * Change the given URL path's file extension to a different one. `newExt`
+ * should not include the leading ".".
+ */
+export const changeFileExtension = (path: string, newExt: string): string => {
```

Comment:
I am not certain of the visibility and thus defensiveness required for this function, but should we throw on empty strings?

`changeFileExtension('test', '') // "test."`

`changeFileExtension('', 'ts') // ".ts"`

Reply:
The visibility is just our worker, and I think we don't need to throw because we're only going to be calling this for `.js` files, and the behavior we have is kinda reasonable, even though it would be a weird input.